### DISCUSSION TITLE
Restricted instances not sending MIDI from LUA in Windows

### DISCUSSION THREAD

**Discussion by Godlike-Productions** [RomanKubiak/ctrlr#289] [Q&A]
When I launch my panel within the ctrlr plugin, it works fine. As soon as I render it as a restricted instance all of the MIDI activity from LUA fails to work.

I've tried both ctrlr 6.0.26 and 5.4.29..

Has anyone seen this activity and know a fix?

**Comment by Godlike-Productions** [comment]
5.3.201 does work, apart from a mass of errors associated with the file operations.  I really want to avoid having to make it fit in that version. Not even sure if it supports json.

**Reply by dobo365** [reply]
Well... Not my experience since several years.
5.3.201 works fine for me even with file operations that I'm using in all my panels.
Just use that version and forget all the other ones (for the moment)

**Reply by Godlike-Productions** [reply]
Thanks. I'll have to spend some more time bashing it into shape for this version.

I'll post a few specific questions seperately.

**Comment by dnaldoog** [comment]
Is this related?

https://ctrlr.org/forums/topic/standalone-images-not-loading/

**Reply by Godlike-Productions** [reply]
Ugh, possibly.

Really weird thing.  The VST runs fine on the computer I exported it from. LUA seems to be disabled on my studio PC.  This is super weird.  I'll have a look at this tomorrow when I have some energy.

I downgraded it to v5.3.201, but I'll go through this post in detail to see if I can sort it out.

Really frustrating.

**Reply by Godlike-Productions** [reply]
I think I've sorted it.  The example panel you posted doesn't work.  Neither of the strings shows up on panel load.

I just put a 400ms timer at the end of the loadedPanel method to set bPanelLoaded to true.  Seems to have fixed everything.

Hopefully finalizing the elimination of the bugs, especially now that it's all released and all.
